LibWeb: Implement missing parent table fixup

kalenikaliaksandr opened the pull request.

martinfalisse commented on Userland/Libraries/LibWeb/Layout/TreeBuilder.cpp:
```diff
@@ -416,6 +416,11 @@ static bool is_not_proper_table_child(Node const& node)
     return !is_table_track_group(display) && !is_table_track(display) && !display.is_table_caption();
 }
 
+static bool is_proper_table_child(Node const& node)
```
These functions seem a bit much, don't you think?

I think it'd be cleaner to remove the is_not_proper_table_child and refactor that functionality to the is_proper_table_child function. (And the same for the other is_not_table_row, is_not_table_cell)

martinfalisse replied:
I actually meant removing the is_not_proper_table_child function, and replacing its use with a `!is_proper_table_child`. And the same for the others.

Perhaps this could be done in a separate commit.

kalenikaliaksandr replied:
hm, but `for_each_sequence_of_consecutive_children_matching` expects function as argument so there would have to be `is_not_` functions anyway, right?

martinfalisse replied:
True, I missed that. Now that you've done the refactor though I would maybe split that into a separate commit (or revert it). Sorry about that.

martinfalisse reviewed (commented):
The commit message doesn't really mean anything to me. Could you be more precise, please?

kalenikaliaksandr commented:
> The commit message doesn't really mean anything to me. Could you be more precise, please?
I added link to doc and number of sections that I implemented.
